### PR TITLE
Create Organization site Admin on Organization Create

### LINKED ISSUE #931: Add Facility Administrator Section to Create Organization Form
Add a new section to the create organization form [`/admin/create-organization`]

Mock:
Screen Shot 2021-03-17 at 9 57 59 AM

Frontend should call the `createOrganization` mutation with the additional fields
```
adminFirstName: String!
adminLastName: String!
adminEmail: String!
adminWorkPhone: String!
adminCellPhone: String!
```

### LINKED ISSUE #1108: Create Organization site Admin on Organization Create
Add the following fields to the `createOrganization` mutation
```
adminFirstName: String!
adminLastName: String!
adminEmail: String!
adminWorkPhone: String!
adminCellPhone: String!
```

update the createOrganization call to also create and creat a new okta user with the new admin fields and add them to the ADMIN group

### PR DESCRIPTION
**Pull request by ClydeDroid**
## Related Issue or Background Info

- Resolves #1108 
- Resolves #931

## Changes Proposed

- Changes `createOrganization` mutation to include admin user info
- Adds a section for admin info to the create org form on the frontend
- I also changed `start.sh` to load env vars in `.env` so that I could get the form to submit (needed smarty streets token)
- Refactored the facility form validation into a reusable hook `useFacilityValidation`

## Additional Information

- The linked issues originally called for accepting phone numbers for the admin, but @aliciabeckett-gov confirmed we don't need to store these, so I changed the accepted info to match the parameters from the `createUser` function in the `ApiUserService` (first middle last suffix and email)

## Screenshots / Demos
image

### CODE REVIEW

File: frontend/src/app/admin/Organization/FacilityAdmin.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import React from "react";
+
+import TextInput from "../../commonComponents/TextInput";
```
Comment (timothybest-usds) [inline frontend/src/app/admin/Organization/FacilityAdmin.tsx:3]:
```suggestion
import Input from "../../commonComponents/Input";
```

File: frontend/src/app/admin/Organization/FacilityAdmin.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  admin: FacilityAdmin;
+  updateAdmin: (admin: FacilityAdmin) => void;
+}
+
+const FacilityAdmin: React.FC<Props> = ({ admin, updateAdmin }) => {
+  const onChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
+  ) => {
+    updateAdmin({ ...admin, [e.target.name]: e.target.value });
+  };
```
Comment (timothybest-usds) [inline frontend/src/app/admin/Organization/FacilityAdmin.tsx:15]:
```suggestion
    const onChange = <K extends keyof FacilityAdmin>(field: K) => (
    value: FacilityAdmin[K]
  ) => {
    updateAdmin({ ...admin, [field]: value });
  };
  ```

File: frontend/src/app/admin/Organization/FacilityAdmin.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      <div className="usa-card__body">
+        <div className="grid-row grid-gap">
+          <div className="tablet:grid-col">
+            <TextInput
+              label="First name"
+              name="firstName"
+              value={admin.firstName || ""}
+              onChange={onChange}
+              required
+            />
```
Comment (timothybest-usds) [inline frontend/src/app/admin/Organization/FacilityAdmin.tsx:31]:
```suggestion
            <Input
              label="First name"
              field="firstName"
              formObject={admin}
              onChange={onChange}
              required
            />
```

Comment (timothybest-usds) [inline frontend/src/app/admin/Organization/FacilityAdmin.tsx:31]:
```suggestion
            <Input
              label="First name"
              field="firstName"
              formObject={admin}
              onChange={onChange}
              required
            />
```

File: frontend/src/app/admin/Organization/FacilityAdmin.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+          </div>
+          <div className="tablet:grid-col">
+            <TextInput
+              label="Middle name"
+              name="middleName"
+              value={admin.middleName || ""}
+              onChange={onChange}
+            />
```
Comment (timothybest-usds) [inline frontend/src/app/admin/Organization/FacilityAdmin.tsx:39]:
```suggestion
            <Input
              label="Middle name"
              field="middleName"
              formObject={admin}
              onChange={onChange}
              required
            />
```

File: frontend/src/app/admin/Organization/FacilityAdmin.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+          </div>
+          <div className="tablet:grid-col">
+            <TextInput
+              label="Last name"
+              name="lastName"
+              value={admin.lastName || ""}
+              onChange={onChange}
+              required
+            />
```
Comment (timothybest-usds) [inline frontend/src/app/admin/Organization/FacilityAdmin.tsx:48]:
```suggestion
            <Input
              label="Last name"
              field="lastName"
              formObject={admin}
              onChange={onChange}
              required
            />
```

File: frontend/src/app/admin/Organization/FacilityAdmin.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+          </div>
+          <div className="tablet:grid-col">
+            <TextInput
+              label="Suffix"
+              name="suffix"
+              value={admin.suffix || ""}
+              onChange={onChange}
+            />
```
Comment (timothybest-usds) [inline frontend/src/app/admin/Organization/FacilityAdmin.tsx:56]:
```suggestion
            <Input
              label="Suffix"
              field="suffix"
              formObject={admin}
              onChange={onChange}
            />
```

Comment (timothybest-usds) [inline frontend/src/app/admin/Organization/FacilityAdmin.tsx:56]:
```suggestion
            <Input
              label="Suffix"
              field="suffix"
              formObject={admin}
              onChange={onChange}
            />
```

File: frontend/src/app/admin/Organization/FacilityAdmin.tsx [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+        </div>
+        <div className="grid-row grid-gap">
+          <div className="tablet:grid-col">
+            <TextInput
+              label="Email"
+              name="email"
+              value={admin.email}
+              onChange={onChange}
+              required
+            />
```
Comment (timothybest-usds) [inline frontend/src/app/admin/Organization/FacilityAdmin.tsx:67]:
```suggestion
            <Input
              label="Email"
              field="email"
              formObject={admin}
              onChange={onChange}
              required
            />
```

**Review by timothybest-usds — CHANGES_REQUESTED** [review]
I added a new component `Input`. Which is a strictly typed wrapper around `TextInput` that also stop the change event from leaking into the form logic. Added some code suggestions to use this new component in `FacilityAdmin.tsx`

**Comment by maya** [comment]
I'm not able to view this locally bc of the following typescript error:

```
/Users/mayabenari/Code/prime-simplereport/frontend/src/app/admin/Organization/FacilityAdmin.tsx
TypeScript error in /Users/mayabenari/Code/prime-simplereport/frontend/src/app/admin/Organization/FacilityAdmin.tsx(29,15):
Type '<K extends "email" | "firstName" | "middleName" | "lastName" | "suffix">(field: K) => (value: FacilityAdmin[K]) => void' is not assignable to type '(field: ReactText) => (value: string) => void'.
  Types of parameters 'field' and 'field' are incompatible.
    Type 'ReactText' is not assignable to type '"email" | "firstName" | "middleName" | "lastName" | "suffix"'.
      Type 'string' is not assignable to type '"email" | "firstName" | "middleName" | "lastName" | "suffix"'.  TS2322

    27 |               field="firstName"
    28 |               formObject={admin}
  > 29 |               onChange={onChange}
       |               ^
    30 |               required
    31 |             />
    32 |           </div>
```

If I'm just reviewing by the screenshot, then it's not up-to-date with the current single-column design added in #1243 and will need to be updated to follow.

**Review by maya — CHANGES_REQUESTED** [review]
See: https://github.com/CDCgov/prime-simplereport/pull/1241#issuecomment-815358145

Assuming this PR is changes to the facility form.

**Comment by ClydeDroid** [comment]
@maya the typescript error was introduced by Tim's suggested changes, and I am working on resolving that.
The screenshot was taken before #1243 was merged. I have merged main to this branch since then, so the changes should be reflected, and you will be able to see that once I sort out the error.

**Comment by ClydeDroid** [comment]
@maya Okay, changed the facility admin section to only have one column, does this look right?
image

**Comment by maya** [comment]
Better! Though do we know where org info is coming from? b/c that appears to be in 2 columns instead of 1 in the screenshot

**Comment by ClydeDroid** [comment]
I can update that as well

**Comment by ClydeDroid** [comment]
image

**Review by maya — APPROVED** [review]
Front end LGTM. Added a few minor type and spacing updates:
 
localhost_3000_admin_create-organization_facility=1b2416eb-2c82-4e21-80ca-2f398365ce7b

**Review by timothybest-usds — APPROVED** [review]

**Pull request merged**